### PR TITLE
user to multiple space tested and implemented

### PR DESCRIPTION
**Pull request by sachinkaria**
connects #10

### CODE REVIEW

File: app/app.rb [hunk 1]
```diff
@@ -6,15 +6,17 @@
...
   end
 
   post '/space/new' do
-    @space = Space.create(name: params[:name],
-                         description: params[:description],
-                         price: params[:price])
+    user = User.get(session[:user_id])
+    user.spaces.create(name: params[:name],
+                 description: params[:description],
+                 price: params[:price])
```
Comment (junyuanxue) [inline app/app.rb]: Alignment baby!

File: app/app.rb [hunk 2]
```diff
@@ -28,6 +30,21 @@ class MakersBnb < Sinatra::Base
...
 
+  get '/user/new' do
+    erb(:"user/new")
+  end
+
+  post '/user/new' do
+    @user = User.create(name: params[:name])
+    session[:user_id] = @user.id
+    redirect '/space/new'
+  end
```
Comment (junyuanxue) [inline app/app.rb:43]: do we need instance var here? (use local var instead of instance var where possible)

File: app/models/space.rb [hunk 3]
```diff
@@ -8,4 +9,6 @@ class Space
   property :description, String
   property :price,       Float, :scale => 2
```
Comment (junyuanxue) [inline app/models/space.rb]: use `scale: 2`

File: spec/features/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      register_user
+      register_space
+      register_space(name: "second space")
+      expect(Space.count).to eq 2
+    end
+
+    scenario 'users spaces are shown on space index' do
+      register_user
+      register_space
+      register_space(name: "second space")
```
Comment (junyuanxue) [inline spec/features/user_spec.rb]: can refactor 

```
+      register_user
 +      register_space
 +      register_space(name: "second space")
```

using `before` or a helper method?

**Comment by junyuanxue** [comment]
:shipit:

**Comment by hawksdoves** [comment]
:shipit:

**Pull request merged**